File: OOps/class.ts
class Car {
  make: string;
  model: string;
  year: number;

  constructor(make: string, model: string, year: number) {
    this.make = make;
    this.model = model;
    this.year = year;
  }

  startEngine(): void {
    console.log(`Starting engine for ${this.make} ${this.model} (${this.year})`);
  }
}

const toyotaCar = new Car("Toyota", "Camry", 2022);
toyotaCar.startEngine();

const chevroletCar = new Car("Chevrolet", "Camaro", 2023);
chevroletCar.startEngine();

File: OOps/class.py
class Car:
    def __init__(self, make, model, year):
        self.make = make
        self.model = model
        self.year = year
    
    def start_engine(self):
        print(f"Starting engine for {self.make} {self.model} ({self.year})")
        

toyota_car = Car("Toyota", "Camry", 2022)
toyota_car.start_engine()

chevrolet_car = Car("Chevrolet", "Camaro", 2023)
chevrolet_car.start_engine()